### PR TITLE
Add sphinx based documentation

### PR DESCRIPTION
**Pull request by Nicoretti**

### CODE REVIEW

**Comment by ckunki** [comment]
The following commits look fine: 
* Add exasol-toolbox as development dependency - commit 9af28476553535f83d1791e7ff6446cb27e7b018  
* Add Sphinx-based documentation skeleton - commit 5983c4e6dd716bd506a78375f411a288401afbaa 
* Convert changelog for Sphinx-based documentation - commit 22521bf051c0118a581a3c1b9a882f56b2b4b261 
* Remove multiversion build - commit 9c5a22916ec6ba5a979bfbec8fe2663e05e33bba 
* Add best practices to the user guide - commit 7c04e9fca85138a76d7b60a94f5fc46f2072c18f 
* Add DBAPI Compatibility Documentation - commit 8d0d11e3bdff1d5aa126e1733e479911739a6216 
* Add documentation for dependencies - commit 4ce0665c86c63991e26bca131c48150594afb652
* Rename dbapi documentation - commit 5f2278822b02203f874a85fa0a2dff7843ac52d9
* Add documentation regarding design decisions. - commit be535ea1e4301bd044257ef9a138cc6179c9429d
* Convert the developer guide to RST format - commit 61ebeb288f51700760c7086f3c09e2ea1403714a
* Convert encryption documentation to RST - commit c36ea4f09fae11d948dc4c4971596e286d694ae0
*  Convert Examples Documentation to ReST - commit 447a8e9865ba315cedd91f35cd505c153bfa53b4
* Reorder the index for the user guide - commit 3bfb914fc6cd9d853de2c3b01c0b229da5652c7b

Note: For the RST format documents resulting from converting and renaming the old documents in markdown format I did not thoroughly review contents, typos, style, grammar or formatting. 

I propose to do that in a seconds step after the conversion is completed.

**Comment by ckunki** [comment]
I additionally reviewed the following commits.

Note again: For the RST format documents resulting from converting and renaming the old documents in markdown format I did not thoroughly review contents, typos, style, grammar or formatting.

I propose to do that in a seconds step after the conversion is completed.

* Convert HTTP transport documentation to reStructuredText - commit 45ba6f6df4cff8b34c2a1d2872989d518d53ccd6
* Convert http transport parallel to sphinx - commit a969781dfe74a9f1cd0460ae7f9ebefd3fa144be
* Convert Known Issues to reStructuredText - commit 2ceb0530742f0ac4c5a19685376f488dfced50c2
* Convert local configuration documentation to reStructuredText - commit ad79d399f93a01f9731d47c877de321bf46b1db6
* Convert Parallelism Documentation to RST - commit d7bd3e85d23e295935db23dbd992db29ee43db6a
* Convert performance to RST - commit 387ab05f0ab82ec1b9aae3e46a815b8adbdeb8cc
* Convert Protocol Version Documentation to RST - commit ac2082141b02c36a76e228410427d7ab48b2bd7b
* Convert UDF Script Output Documentation to RST - commit bf1830caf434e34f2a49f855fa83001265721df3
* Convert snapshot transaction documentation to RST - commit 32da556631bfef27a721846c3054bf27b4d90169
* Convert SQL formatting documentation to RST - commit 62ba84a5be977c967bf68dc4690dc3e60e87747d
* Fix minor issues like broken links etc. - commit ca225fd062b35a97bc4498704d64faba48feb507

Last not least: There are some Unchanged files with check annotations mentioned in the PR.
Please fix.

**Review by ckunki — DISMISSED** [review]
See comments in PR.

**Comment by Nicoretti** [comment]
> Last not least: There are some Unchanged files with check annotations mentioned in the PR. Please fix.

From what I understand these are fallse, positives.

File: doc/user_guide/best_practices.rst [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+``pip install pyexasol[orjson]``
+
+You may try any other json library. All you need to do is to overload ``_init_json()`` method in ``ExaConnection``.
+
+Use ``.meta`` Functions to Perform Lock-Free Meta Data Requests
+---------------------------------------------------------------
+It is quite common for Exasol system views to become locked by DML statements, which prevents clients from retrieving metadata.
+
+In order to mitigate this problem, Exasol provided special SQL hint described in `IDEA-476 <https://www.exasol.com/support/browse/IDEA-476>` which is available in latest versions. It does not require user to enable "snapshot transaction" mode for the whole session. Currently this is the best way to access meta data using WebSocket protocol.
```
Comment (Jannis-Mittenzwei) [inline doc/user_guide/best_practices.rst:112]: broken link

File: doc/user_guide/dependencies.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Core dependencies
++++++++++++++++++
+
+- Exasol >= 6.2
+- Python >= 3.6
+- websocket-client >= 1.0.1
+- pyopenssl
+- rsa
+
+Optional dependencies
```
Comment (Jannis-Mittenzwei) [inline doc/user_guide/dependencies.rst:13]: Highlight optional dependencies more " ``dependency name`` " for more clarity

File: doc/user_guide/local_config.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    compression = True
+    fetch_dict = True
+
+You may specify any parameters available in ``connect`` function,
+except parameters expecting a Python class or function.
+
+You may specify multiple sections. Each section represents a separate connection config.
+It might be useful if you have multiple Exasol instances.
+
+In order to create a connection using local config, please call function ``connect_local_config <https://../docs/REFERENCE.md#connect_local_config>``.
```
Comment (Jannis-Mittenzwei) [inline doc/user_guide/local_config.rst:26]: broken link

File: doc/user_guide/udf_script_output.rst [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+SCRIPT MODE
+-----------
+
+Script mode is useful for production usage and during the last stages of development.
+
+Connections are accepted from all VMs. The output of each VM is stored in a separate log file.
+
+The server runs for a single SQL statement and stops automatically.
+
+How to use it:
```
Comment (Jannis-Mittenzwei) [inline doc/user_guide/udf_script_output.rst:44]: missing blank line? Is not recognized as an enumeration.

**Comment by ckunki** [comment]
I additionally reviewed the following commits.

Note again: For the RST format documents resulting from converting and renaming the old documents in markdown format I did not thoroughly review contents, typos, style, grammar or formatting.

I propose to do that in a seconds step after the conversion is completed.

* Fix the invalid reference - commit 5aed80c097a005f15d9b6de414e10114e0001636
* Fix file name extension - commit 099ac3bab4894e361cc231a076c279b39e8ab5d3
* Address the reviewed feedback - commit 376cca89215cc507ff4640bec8cf2ea613ab8894

**Review by ckunki — APPROVED** [review]

**Pull request merged**